federatedscope/core/auxiliaries/utils.py

```diff
@@ -500,3 +500,9 @@ def calculate_time_cost(instance_number,
         comm_cost = 0
 
     return comp_cost, comm_cost
+
+
+def merge_param_dict(raw_param, filtered_param):
+    for key in filtered_param.keys():
+        raw_param[key] = filtered_param[key]
```

Comment:
Maybe only cover the attribute `data`?

Reply:
When the user forgets to filter some certain parameters, let's say `conv.weight`, and the `raw_param` doesn't have the key, maybe it is strange for the user to report KeyError here (Because they don't know why we merge the parameters here). 
So maybe we should raise a detailed error here to explain it?

Reply:
State dict only contains param, and grads are in model. parameters()